tests/distributions/test_multivariate.py

```diff
@@ -13,6 +13,107 @@
+            len_u = tf.shape(u)[0]
+            dst = MultivariateNormalCholesky(tf.zeros([2]), 
+                                         tf.zeros([len_u, len_u]))
+            with self.assertRaisesRegexp(
+                    tf.errors.InvalidArgumentError, 'compatible'):
+                dst.sample().eval(feed_dict={u: np.ones((3,))})
+
+    def test_shape_inference(self):
+        with self.test_session(use_gpu=True):
+            # Static
+            mean = 10 * np.random.normal(size=(10, 11, 2)).astype('d')
+            cov = np.zeros((10, 11, 2, 2))
+            dst = MultivariateNormalCholesky(
+                tf.constant(mean), tf.constant(cov))
+            self.assertEqual(dst.get_batch_shape().as_list(), [10, 11])
+            self.assertEqual(dst.get_value_shape().as_list(), [2])
+            # Dynamic
+            unk_mean = tf.placeholder(tf.float32, None)
+            unk_cov = tf.placeholder(tf.float32, None)
+            dst = MultivariateNormalCholesky(unk_mean, unk_cov) 
+            self.assertEqual(dst.get_value_shape().as_list(), [None])
+            feed_dict = {unk_mean: np.ones(2), unk_cov: np.eye(2)}
+            self.assertEqual(list(dst.batch_shape.eval(feed_dict)), [])
+            self.assertEqual(list(dst.value_shape.eval(feed_dict)), [2])
+
+
+    def _gen_test_params(self):
+        np.random.seed(23)
+        mean = 10 * np.random.normal(size=(10, 11, 3)).astype('d')
+        cov = np.zeros((10, 11, 3, 3))
+        cov_chol = np.zeros_like(cov)
+        for i in range(10):
+            for j in range(11):
+                cov[i, j] = stats.invwishart.rvs(3, np.eye(3))
+                cov[i, j] /= np.max(np.diag(cov[i, j]))
+                cov_chol[i, j, :, :] = np.linalg.cholesky(cov[i, j])
+        return mean, cov, cov_chol
+
+
+    def test_sample(self):
+        with self.test_session(use_gpu=True):
+            mean, cov, cov_chol = self._gen_test_params()
+            dst = MultivariateNormalCholesky(
+                tf.constant(mean), tf.constant(cov_chol))
+            n_exp = 20000
+            samples = dst.sample(n_exp).eval()
+            self.assertEqual(samples.shape, (n_exp, 10, 11, 3))
+            self.assertAllClose(
+                np.mean(samples, axis=0), mean, rtol=5e-2, atol=5e-2)
+            for i in range(10):
+                for j in range(11):
+                    self.assertAllClose(
+                        np.cov(samples[:, i, j, :].T), cov[i, j], 
+                        rtol=1e-1, atol=1e-1)
+
+
+    def test_prob(self):
+        with self.test_session(use_gpu=True):
+            mean, cov, cov_chol = self._gen_test_params()
+            dst = MultivariateNormalCholesky(
+                tf.constant(mean), tf.constant(cov_chol), check_numerics=True)
+            n_exp = 200
+            samples = dst.sample(n_exp).eval()
+            log_pdf = dst.log_prob(tf.constant(samples)).eval()
+            for i in range(10):
+                for j in range(11):
```

Comment:
similar to above

Reply:
It seems that `stats.multivariate_normal.logpdf` only accepts samples from, and parameters of, a single distribution in a call, and we are testing 10*11 distributions here.